src/components/AugmentedResultList/AugmentedResultList.ts

```diff
@@ -0,0 +1,141 @@
+import { ComponentOptions, IComponentBindings, Initialization, IQueryResult, IResultListOptions, IFieldOption } from 'coveo-search-ui';
+
+/**
+ * Interface for the data returned from external fetch action.
+ *
+ * @export
+ * @interface IAugmentData
+ */
+export interface IAugmentData {
+    /**
+     * Data specific to a result with matching object id.
+     */
+    resultData: {}[];
+    /**
+     * Data to add to every result with matching object id.
+     */
+    commonData: {};
+}
+
+/**
+ * Generic interface for the response returned by the external fetch action.
+ *
+ * @export
+ * @interface IPromiseReturnArgs
+ * @template T
+ */
+export interface IPromiseReturnArgs<T> {
+    data: T;
+}
+
+export interface AugmentedResultListOptions extends IResultListOptions {
+    matchingIdField: IFieldOption;
+    fetchAugmentData: (objectIds: String[]) => Promise<IPromiseReturnArgs<IAugmentData>>;
+    matchingFunction: (augmentData: any, queryResult: IQueryResult) => boolean;
+}
+
```

Comment:
Could be nice to add a doc string to link to `[Using the Augmented Result List Component](http://docs.coveo.com/en/3421)`, which will be published when [this PR](https://bitbucket.org/coveord/doc_jekyll-public-site/pull-requests/4748/sfint-3523-augmented-result-list/diff) is merged.

Reply:
I'm not so sure about that one since this component is meant to be agnostic to any specific platform like salesforce.

Reply:
fair enough 🙃